From 5717fde10dcd99605882c06aecb1db31615e30d2 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 27 Jan 2015 17:11:14 +0800
Subject: [PATCH] MLK-10101-4 usb: add otg_fsm pointer in usb_bus

Add otg_fsm pointer in struct of usb_bus for access otg_fsm via bus.
Original way was to put it in usb_otg, then usb host can access otg_fsm via
hcd->usb_phy->otg->fsm, since usb_phy will not be the future direction, instead
phy is prefered, so this way may not work. It's more direct and simple to put
it in usb_bus.

Signed-off-by: Li Jun <jun.li@freescale.com>
---
 include/linux/usb.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/usb.h b/include/linux/usb.h
index 7f6eb85..a7bdc61 100644
--- a/include/linux/usb.h
+++ b/include/linux/usb.h
@@ -329,6 +329,7 @@ struct usb_bus {
 					 * Does the host controller use PIO
 					 * for control transfers?
 					 */
+	struct otg_fsm *otg_fsm;	/* usb otg finite state machine */
 	u8 otg_port;			/* 0, or number of OTG/HNP port */
 	unsigned is_b_host:1;		/* true during some HNP roleswitches */
 	unsigned b_hnp_enable:1;	/* OTG: did A-Host enable HNP? */
-- 
1.8.0

